src/main/java/service/strategy/strategyimpl/PurchaseOperation.java

```diff
@@ -0,0 +1,20 @@
+package service.strategy.strategyimpl;
+
+import dao.Storage;
+import model.FruitRecord;
+
+
+public class PurchaseOperation implements OperationHandler {
+
+    @Override
+    public void apply(FruitRecord transaction) {
+        Integer currentQuantity = Storage.storage.get(transaction.getFruit());
+        int purchaseQuantity = transaction.getQuantity();
+        if (currentQuantity == null && currentQuantity < 0) {
+            throw new IllegalArgumentException("Invalid quantity of for purchase.");
```

Comment:
The exception message 'Invalid quantity of for purchase.' is not informative enough. Consider including details about the fruit and the quantities involved to make it clearer what went wrong.